status/status.go

```diff
@@ -533,3 +576,30 @@ func (s Status) IsPRInSnapshotOpened(ctx context.Context, reporter ReporterInter
 	}
 	return false, err
 }
+
+// getComponentSnapshotsFromGroupSnapshot return the component snapshot list which component snapshot is created from
+func getComponentSnapshotsFromGroupSnapshot(ctx context.Context, c client.Client, groupSnapshot *applicationapiv1alpha1.Snapshot) ([]*applicationapiv1alpha1.Snapshot, error) {
+	var componentSnapshotInfos []*gitops.ComponentSnapshotInfo
+	var componentSnapshots []*applicationapiv1alpha1.Snapshot
+	var err error
+	if componentSnapshotInfoString, ok := groupSnapshot.Annotations[gitops.GroupSnapshotInfoAnnotation]; ok {
+		componentSnapshotInfos, err = gitops.UnmarshalJSON([]byte(componentSnapshotInfoString))
+		if err != nil {
+			return nil, fmt.Errorf("failed to unmarshal JSON string: %w", err)
+		}
+	}
+
+	for _, componentSnapshotInfo := range componentSnapshotInfos {
+		componentSnapshot := &applicationapiv1alpha1.Snapshot{}
+		err := c.Get(ctx, types.NamespacedName{
+			Namespace: groupSnapshot.Namespace,
+			Name:      componentSnapshotInfo.Snapshot,
+		}, componentSnapshot)
+		if err != nil {
+			return nil, fmt.Errorf("failed to find component snapshot %s included in group snapshot %s/%s: %w", componentSnapshotInfo.Snapshot, groupSnapshot.Namespace, groupSnapshot.Name, err)
```

Comment:
I'm thinking if we should log the error here but continue the loop anyways to ensure that the other Snapshot PRs get the report sent to them? I'm not sure if that's the way to go, wdyt?

Reply:
It makes sense, let me update it. Thanks.

Reply:
Updated.